src/odemis/odemisd/modelgen.py

```diff
@@ -376,14 +376,14 @@ def _fill_creator(self):
                 parents = [p for p in parents if self.ast[p].get("class") != "Microscope"]
 
                 if len(parents) == 0:
-                    raise SemanticError("Error in microscope file: component %s "
-                            "has no class specified and is not created by any "
-                            "component." % name)
+                    raise SemanticError("Error in microscope file: component \"%s\" "
+                            "has no class specified and not component declares in its "
+                            "children." % name)
```

Comment:
```suggestion
                    raise SemanticError("Error in microscope file: component \"%s\" "
                            "has no class specified and not component declares in its "
                            "children." % name)
```
This sentence looks a little weird to me, could you rephrase that?

Reply:
How about: `"Error in microscope file: component \"%s\ has no class specified and is not child of any component."` ?

Reply:
yeah that sounds much clearer :)